### PR TITLE
CPP API for SPI

### PR DESCRIPTION
**Pull request by petersonev**

### CODE REVIEW

File: API/cpp/Src/DigitalOut.cpp [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 DigitalOut::DigitalOut(PinName pin, PullType pull, PinMode mode, PinSpeed speed,
         bool inverted) : pin(pin), inverted(inverted) {
-    GPIO_InitTypeDef pin_structure;
+    GPIO_InitTypeDef pin_structure = {};
```
Comment (Arvind-Srinivasan) [inline API/cpp/Src/DigitalOut.cpp]: Why this change, just curious?

Comment (petersonev) [reply API/cpp/Src/DigitalOut.cpp]: That was the issue we were having with SPI so I want to play it safe and zero out all structs like this
